src/lib/ngsiNotify/Notifier.cpp

```diff
@@ -238,7 +238,7 @@ static std::vector<SenderThreadParams*>* buildSenderParamsCustom
     //
     // 3. Payload
     //
-    if (httpInfo.payload.empty())
+    if (httpInfo.includePayload)
```

Comment:
Not sure about this fix. I understand the idea in the original code is something like this:

```
if (http.payload.empty())
{
   <not use any special payload and use the NGSI default one>
}
else
{
   <use the payload from the custom notification>
}
```

but note that in both branches of the if a payload is used in the notification.

Thus, the check for includePayload should be a "higher level". Something like this:

```
if (httpInfo.includePayload)
{
   if (http.payload.empty())
   {
      <not use any special payload and use the NGSI default one>
   }
   else
   {
      <use the payload from the custom notification>
   }
}
else
{
   <not include paylaod at all in the notification (in fact, maybe this else block is not needed at all if there is nothing to do in this case)>
}
```

However, I don't like too much the double-if as it makes readibility worse, from my point of view. Maybe you can find a way of refactoring to improve my suggestion :)

Reply:
updated